### PR TITLE
chore: Add feature-gated tests to CI

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

The goal is that this test suite will be used for feature-gated features that are incoming, e.g. #6198 and #6600 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -420,6 +416,20 @@ const (
 	prometheusOperatorServiceName = "prometheus-operator"
 )
 
+// TestFeatureGatedFeatures tests features that are behind feature gates.
+// Altough we'd like to have these tests green at all times, they are not
+// critical for the release process, therefore they can fail without blocking PRs.
+func TestFeatureGatedFeatures(t *testing.T) {
```
Comment (simonpasquier) [inline test/e2e/main_test.go:422]: (nit)
```suggestion
func TestGatedFeatures(t *testing.T) {
```

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -99,7 +94,7 @@ jobs:
         EXCLUDE_PROMETHEUS_ALL_NS_TESTS=${{ matrix.prometheusAllNS }}
         EXCLUDE_THANOSRULER_TESTS=${{ matrix.thanosruler }}
         EXCLUDE_OPERATOR_UPGRADE_TESTS=${{ matrix.operatorUpgrade }}
-        FEATURE_GATED_TESTS=${{ matrix.featureGated }}
+        FEATURE_GATED_TESTS=exclude
```
Comment (simonpasquier) [inline .github/workflows/e2e.yaml:97]: should we rename to EXCLUDE_FEATURE_GATED_TESTS for consistency?

File: test/e2e/main_test.go [hunk 3]
```diff
@@ -420,6 +416,20 @@ const (
 	prometheusOperatorServiceName = "prometheus-operator"
 )
 
+// TestFeatureGatedFeatures tests features that are behind feature gates.
+// Altough we'd like to have these tests green at all times, they are not
+// critical for the release process, therefore they can fail without blocking PRs.
```
Comment (simonpasquier) [inline test/e2e/main_test.go:421]: I might not agree 100% with this :)

For now, all feature gates are alpha so yes they could break anytime but in the future, some of them (hopefully) will become beta. Past experience also showed me that non-gating jobs are very likely to degrade over time.

**Review by simonpasquier — COMMENTED** [review]
Thanks for starting this!

File: .github/workflows/e2e.yaml (hunk 2 above)
Comment (ArthurSens) [reply .github/workflows/e2e.yaml:97]: yeah good idea

File: test/e2e/main_test.go (hunk 3 above)
Comment (ArthurSens) [reply test/e2e/main_test.go:421]: Fair enough, I did not think of beta graduation of features.

Should we split tests again in the future? (Beta are required to pass, Alpha not required)

Comment (simonpasquier) [reply test/e2e/main_test.go]: We can still merge without all required tests being green so I'd rather be strict from the start and avoid considering these tests as "less important".

Comment (ArthurSens) [reply test/e2e/main_test.go]: Fair point, changes adjusted

**Review by simonpasquier — APPROVED** [review]
This is good enough to get started!

**Pull request merged**